### PR TITLE
Propose "why" folder that we will keep as "advisors" knowledge base

### PR DESCRIPTION
**Pull request by potiuk**
I would like to propose the "why" folder, where we will keep a (very) short summary about some of the rules and policies that we will be - as Advisors - discussing with the projects we will advise.

I think we need something like that - also as a place where we
- advisors - will be able to update and exchange our understanding on why there are certain rules and policies. Also to discuss future variations and changes to those rules, possibly resulting from interactions with multiple PMCs when we will learn from them how they are applying certain rules and policies in their environment.

I gave an example of how I see such "why" captured information could look like - starting with the most familiar (for me) release process and I am curious what others think about it.

### CODE REVIEW

**Review by rbowen — APPROVED** [review]
Interesting idea. Yeah, I like this. It also gives us something to point to when projects push back, in a "this isn't personal, it's just the way things are" kind of way. Especially in conjunction with the notion that we're also there to learn, and possibly improve the process/rules.

**Comment by potiuk** [comment]
> Interesting idea. Yeah, I like this. It also gives us something to point to when projects push back, in a "this isn't personal, it's just the way things are" kind of way. Especially in conjunction with the notion that we're also there to learn, and possibly improve the process/rules.

100% agree. There is a huge value in something being publicly stated in "semi-official" way. Just writing things down and pointing to it as a link is enough to stop treating things as personal opinion but something that is "commonly understood". Also when we connect it with the gorup of advisors who are mostly a "group ASF members who actually care enough to spend their personal time on helping other projects" has additional benefit that it is not seen as something that is "top-> bottom" but more like "group of peers agreed this is how it is".

File: wg-advisors/why/why_release_process.md
Comment (justinmclean) [inline wg-advisors/why/why_release_process.md:22]: I'd note it also gives the PMC and release manager some legal protection.

File: wg-advisors/why/why_release_process.md
Comment (tisonkun) [reply wg-advisors/why/why_release_process.md:30]: > some projects consider the RM as an implicit +1

Here is a patch to explain no RM implicit +1 https://github.com/apache/www-site/pull/338.

File: wg-advisors/red-flags.md
Comment (tisonkun) [inline wg-advisors/red-flags.md:18]: The answer is not included in the new file. Do you expect elaborate it in the new file later, or we should create a dedicated answer.

I remember that the original purpose of this "red flag" was to encourage RMs to waiting for more inputs instead of eagerly cancelling release candidates which increases the release burden.

File: wg-advisors/why/why_release_process.md
Comment (potiuk) [reply wg-advisors/why/why_release_process.md:22]: Good point.

File: wg-advisors/why/why_release_process.md
Comment (potiuk) [reply wg-advisors/why/why_release_process.md:30]: Yes. Worth mentioning the variations.

But I think going into oo many details is not a good idea. It will be of coure always an art how "deep" we should describe it here, but the rule of thumb I think such document should not go into details where the linked document cover them, it should be a one-pager max and it should mostly focus on the most important "why's" - providing a very broad context that is otherwise distiributed among several documents, heads of people, discussions etc. etc. 

I think it serves two purposes:

* for the advisor - so that they can always come back here and immediately see - hey, why actually we are doing it ?  - a... here it is - and go into details if needed

* for the advised ones - to be assured that what we are advising them is not a "cargo cult" and tha we have very good reason that you can read about it here why we are doing it - without having to go through all the piles of documents and discussions or asking others. 

I will expand it a little bit soon.

File: wg-advisors/why/why_release_process.md
Comment (potiuk) [reply wg-advisors/why/why_release_process.md:26]: Yep. I will add it.

File: wg-advisors/red-flags.md
Comment (potiuk) [reply wg-advisors/red-flags.md:18]: It's an interesting one. This is precisely the kind of discussions that I **hoped** the `why` document will start :)

IMHO that one actually deserves a separate "why", as this red flag is (now that you explained it not really about software releases "per se" but about communication and process around the releases, and possibly about involving the community during the relase process and engaging them in testing  - whcih is a subtly different thing and possibly deserves another "why". I wil propose something soon :)

File: wg-advisors/why/why_release_process.md
Comment (potiuk) [reply wg-advisors/why/why_release_process.md]: I added a short description describing those variations - with the goal to make it clearer for both "purposes". I hope it's better now.

File: wg-advisors/why/why_release_process.md
Comment (potiuk) [reply wg-advisors/why/why_release_process.md]: Added. Also split out the "non-negotiables".

File: wg-advisors/red-flags.md
Comment (potiuk) [reply wg-advisors/red-flags.md]: I tried to separate it, but indeed it seems better if it is combined in the release process. I just stressed how important it is to engage the dev community in the release/testing process and added a little description summarising why we are doing it. Should be explained now.

**Comment by potiuk** [comment]
Ready for the next round of reviews.

**Comment by potiuk** [comment]
If there are no more comments. I'd love to get this one merged and then maybe others (I can do as well) add a bit more why`s for different areas. I think also - when we have a few whys we might actually regroup the "red-flags" and have them groupped according to the `why` they are referring to - rather than link them individually - but taht migh be a refactor of the docs when we add 2nd or 3rd why as well.

**Pull request merged**
